test/e2e/commands/xmlupload/utils.py

```diff
@@ -8,18 +8,29 @@
 from rdflib import Graph
 from rdflib import Literal
 
+from dsp_tools.utils.rdflib_constants import KNORA_API
 from dsp_tools.utils.rdflib_constants import SubjectObjectTypeAlias
 
 
 def util_request_resources_by_class(resclass_iri: str, auth_header: dict[str, str], project_iri: str, creds) -> Graph:
+    all_resources = Graph()
+    for i in range(100):
+        g = _get_single_graph_from_api(resclass_iri, auth_header, project_iri, creds, i)
+        all_resources += g
+        if not list(g.triples((None, KNORA_API.mayHaveMoreResults, None))):
+            break
```

Comment:
Going until page number 100 is probably good enough, but not so elegant. What about this?

```suggestion
    mayHaveMoreResults = True
    i = 0
    while mayHaveMoreResults:
        g = _get_single_graph_from_api(resclass_iri, auth_header, project_iri, creds, i)
        all_resources += g
        mayHaveMoreResults = bool(list(g.triples((None, KNORA_API.mayHaveMoreResults, None))))
        i += 1
```

Reply:
I dislike writing `while True` loops.

Reply:
It's not `while True`, it's `while <var>`. Me too, I rarely use while loops, but here it would be a good fit. The for loop isn't a clean solution here. However, it's not super important, so feel free to follow your preferences ;-)